website/source/docs/providers/aws/r/lambda_permission.html.markdown

```diff
@@ -27,14 +27,14 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
 resource "aws_lambda_alias" "test_alias" {
   name             = "testalias"
   description      = "a sample description"
-  function_name    = "${aws_lambda_function.test_lambda.arn}"
+  function_name    = "${aws_lambda_function.test_lambda.function_name}"
   function_version = "$LATEST"
 }
 
 resource "aws_lambda_function" "test_lambda" {
   filename      = "lambdatest.zip"
   function_name = "lambda_function_name"
-  role          = "${aws_iam_role.iam_for_lambda.arn}"
+  role          = "${aws_iam_role.iam_for_lambda.name}"
```

Comment:
Yes, you are right.

I found it on the aws-sdk-go. 

```golang
     // The Amazon Resource Name (ARN) of the IAM role that Lambda assumes when it
     // executes your function to access any other Amazon Web Services (AWS) resources.
     Role *string `type:"string"`

```

https://github.com/aws/aws-sdk-go/blob/6d13e167d1db68c8ef8c1b27066bcd03457d22e3/service/lambda/api.go#L3940-L3942

I'll return here to `arn`.